src/librustc_mir/interpret/machine.rs

```diff
@@ -318,4 +318,11 @@ pub trait Machine<'mir, 'tcx>: Sized {
         _mem: &Memory<'mir, 'tcx, Self>,
         _ptr: Pointer<Self::PointerTag>,
     ) -> InterpResult<'tcx, u64>;
+
+    fn thread_local_alloc_id(
+        _ecx: &mut InterpCx<'mir, 'tcx, Self>,
+        did: DefId,
+    ) -> InterpResult<'tcx, AllocId> {
+        throw_unsup!(ThreadLocalStatic(did))
```

Comment:
I thought about errors in these default implementations again. Currently, we are inconsistent.

`int_to_ptr` and `abort` (and with this PR `thread_local_alloc_id`) have default implementations that error "unsupported".

`binary_ptr_op`, `box_alloc`, `ptr_to_int` also error in both const-eval and const-prop, but with separately implemented error messages.

I am not sure what's better, but it seems odd not to be consistent here.